services/connectivity/messaging/src/main/java/org/eclipse/ditto/services/connectivity/messaging/httppush/HttpPublisherActor.java

```diff
@@ -161,6 +148,35 @@ static Props props(final Connection connection, final HttpPushFactory factory, f
         return Props.create(HttpPublisherActor.class, connection, factory, clientId);
     }
 
+    private Flow<Pair<HttpRequest, HttpPushContext>, Pair<Try<HttpResponse>, HttpPushContext>, ?>
+    buildHttpRequestFlow(final HttpPushConfig config) {
+
+        final Duration requestTimeout = config.getRequestTimeout();
+
+        final PreparedTimer timer = DittoMetrics.timer("http_publish_request_time")
+                // Set maximum duration higher than request timeout to avoid race conditions
+                .maximumDuration(requestTimeout.plus(Duration.ofSeconds(5)))
+                .tag("id", connection.getId().toString());
+
+        final Sink<Duration, CompletionStage<Done>> logRequestTimes = Sink.<Duration>foreach(duration -> {
+            connectionLogger.success("HTTP request took <{0}> ms.", duration.toMillis());
+        });
+
+        return DittoFlowEnhancement
+                .enhanceFlow(factory.<HttpPushContext>createFlow(getContext().getSystem(), logger))
```

Comment:
I'm not really sure what you mean by

>  I would also suggest, in case of a timeout, to just emit failure within Try<HttpResponse> to stay aligned with the current control- and error-flow instead of building yet another flow.

Could you provide a sample implementation?

Reply:
> 
> 
> Have you also considered the client timeout: [Connection Lifetime timeout](https://doc.akka.io/docs/akka-http/current/common/timeouts.html#client-timeouts)?
> From Akka docs:
> 
> > This timeout configures a maximum amount of time, while the connection can be kept open. This is useful, when you reach the server through a load balancer and client reconnecting helps the process of rebalancing between service instances.

No I didn't. I think we should consider to configure this as well.